pkg/components/azure-arc/image/script.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Type of action: register or remove the Lokomotive cluster from Azure Arc.
+action=$1
+
+ca="$(cat /var/run/secrets/kubernetes.io/serviceaccount/ca.crt | base64 | tr -d "\n")"
+namespace="$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)"
+token="$(cat /var/run/secrets/kubernetes.io/serviceaccount/token | tr -d "\n")"
+server="https://kubernetes.default.svc"
+
+echo "
+apiVersion: v1
+kind: Config
+clusters:
+- name: default-cluster
+  cluster:
+    certificate-authority-data: ${ca}
+    server: ${server}
+contexts:
+- name: default-context
+  context:
+    cluster: default-cluster
+    namespace: ${namespace}
+    user: default-user
+current-context: default-context
+users:
+- name: default-user
+  user:
+    token: ${token}
+" > sa.kubeconfig
+
+function azure_login() {
+  az login --service-principal -u ${AZURE_APPLICATION_CLIENT_ID} -p ${AZURE_APPLICATION_PASSWORD} --tenant ${AZURE_TENANT_ID}
+}
+
+function perform_action() {
+  case ${action} in
+    register)
+      # Connect the cluster to Azure Arc
+      az connectedk8s connect --kube-config sa.kubeconfig --name ${CONNECTED_CLUSTER_NAME} --resource-group ${AZURE_RESOURCE_GROUP}
+      ;;
+    remove)
+      # Remove the cluster from Azure Arc
+      az connectedk8s delete --yes --kube-config sa.kubeconfig --name ${CONNECTED_CLUSTER_NAME} --resource-group ${AZURE_RESOURCE_GROUP}
```

Comment:
Should we be verifying that all required environment variables are set and not empty?

Reply:
not sure if it adds value

We check right from the configuration side that its not empty.

Reply:
Given that we do not test values.yaml templating, I think this together with e2e tests (installing component on CI) would give us a bit better test coverage, in case someone messes up Helm chart/values template + it's also make it safer when you run script by hand.